pkg/vault/contracts/Vault.sol

```diff
@@ -1201,27 +1203,25 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
                 // The amount of underlying tokens to deposit is the necessary amount to fulfill the trade
                 // (amountInUnderlying), plus the amount needed to leave the buffer rebalanced 50/50 at the end
                 // (bufferUnderlyingSurplus).
-                vaultUnderlyingDeltaHint = amountInUnderlying + bufferUnderlyingSurplus;
+                vaultUnderlyingDeltaHint = (amountInUnderlying.toInt256() + bufferUnderlyingSurplus).toUint256();
```

Comment:
I don't quite understand why there is `int256` + `uint256`

Reply:
Hey Elshan! Good point, I'll comment it better. bufferUnderlyingSurplus is an int256 that can be a positive or a negative number. So, we need to convert amountINUnderlying to int256, sum both values, then convert again to uint256 because vaultUnderlyingDeltaHint needs to be a positive number (and so does amountInUnderlying)